### ISSUE TITLE
Is name hyphen normalization documented anywhere?

### ISSUE THREAD

**Issue by rtpg** [nushell/nushell#10815]
### Question

A recent shell session:

```
~> def foo [--hi-there] {                                                        10/23/2023 03:01:34 PM
::: $hi-there
::: }; foo
Error: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[entry #40:1:1]
 1 │ def foo [--hi-there] {
 2 │ $hi-there
   · ────┬────
   ·     ╰── expected valid variable name
 3 │ }
   ╰────

~> def foo [--hi-there] {                                                        10/23/2023 03:01:45 PM
::: $hi_there
::: }; foo
false
```

I was trying to follow along in the `flags` section of the docs, and it took me a while to find that names are meant to be underscored. Given that the parser seems to look at the entire name, might be worth it to (for example) pipe the invalid name through the normalizer and see what shows up. Or at least add some docs closer to flags (I might have totally missed it though)

**Comment by WindSoilder** [comment]
Sorry we didn't document it.
We want users to be able to use custom command flags like "foo --hi-there" normally.
But in case of variable usage, since we are not allowed to have variables like "$hi-there", we have to use "$hi_there" as a workaround.

**Comment by sholderbach** [comment]
Good call! worth documenting with the rest of the introduction into command signatures in https://www.nushell.sh/book/custom_commands.html#flags

At the same time yes we could improve the error message for that by doing a "did you mean" lookup with `-` converted to `_`.

**Comment by NotTheDr01ds** [comment]
This was documented in https://github.com/nushell/nushell.github.io/pull/962 (Thanks @tminich!)

Leaving the `unhelpful error` issue open, but removing `documentation` label.

**Comment by rtpg** [comment]
I just made a PR that... doesn't exactly resolve everything but at leasts offers the "right" variable name in the case that you would hit this (#14001). It's a bit silly but sent it in if only to add some tests.

**Comment by WindSoilder** [comment]
Closing the issue because the question is answered.
